src/mainboard/protectli/vault_adl_p/gpio.c

```diff
@@ -0,0 +1,631 @@
+	/* GPD4 - SLP_S3# */
+	PAD_CFG_NF(GPD4, NONE, PWROK, NF1),
+	/* GPD5 - SLP_S4# */
+	PAD_CFG_NF(GPD5, NONE, PWROK, NF1),
+	/* GPD6 - SLP_A# */
+	PAD_CFG_NF(GPD6, NONE, PWROK, NF1),
+	/* GPD7 - GPIO */
+	PAD_NC(GPD7, NONE),
+	/* GPD8 - SUSCLK */
+	PAD_CFG_NF(GPD8, NONE, PWROK, NF1),
+	/* GPD9 - GPIO */
+	PAD_NC(GPD9, NONE),
+	/* GPD10 - SLP_S5# */
+	PAD_CFG_NF(GPD10, NONE, PWROK, NF1),
+	/* GPD11 - GPIO */
+	PAD_NC(GPD11, NONE),
+	/* GPD_INPUT3VSEL - GPD_INPUT3VSEL */
+	PAD_CFG_NF(GPD_INPUT3VSEL, NONE, PWROK, NF1),
+	/* GPD_SLP_LANB - GPD_SLP_LANB */
+	PAD_CFG_NF(GPD_SLP_LANB, NONE, PWROK, NF1),
+	/* GPD_SLP_SUSB - GPD_SLP_SUSB */
+	PAD_CFG_NF(GPD_SLP_SUSB, NONE, PWROK, NF1),
+	/* GPD_WAKEB - GPD_WAKEB */
+	PAD_CFG_NF(GPD_WAKEB, NONE, PWROK, NF1),
+	/* GPD_DRAM_RESETB - GPD_DRAM_RESETB */
+	PAD_CFG_NF(GPD_DRAM_RESETB, NONE, PWROK, NF1),
+
+	/* ------- GPIO Community 4 ------- */
+
+	/* ------- GPIO Group GPP_C ------- */
+
+	/* GPP_C0 - SMBCLK */
+	PAD_CFG_NF(GPP_C0, NONE, DEEP, NF1),
+	/* GPP_C1 - SMBDATA */
+	PAD_CFG_NF(GPP_C1, NONE, DEEP, NF1),
+	/* GPP_C2 - GPIO */
+	PAD_CFG_GPO(GPP_C2, 0, DEEP),
+	/* GPP_C3 - SML0CLK */
+	PAD_CFG_NF(GPP_C3, NONE, DEEP, NF1),
+	/* GPP_C4 - SML0DATA */
+	PAD_CFG_NF(GPP_C4, NONE, DEEP, NF1),
+	/* GPP_C5 - GPIO */
+	PAD_CFG_GPO(GPP_C5, 0, DEEP),
+	/* GPP_C6 - SML1CLK */
+	PAD_CFG_NF(GPP_C6, NONE, RSMRST, NF1),
+	/* GPP_C7 - SML1DATA */
+	PAD_CFG_NF(GPP_C7, NONE, RSMRST, NF1),
+	/* GPP_C8 - GPIO */
+	PAD_NC(GPP_C8, NONE),
+	/* GPP_C9 - GPIO */
+	PAD_NC(GPP_C9, NONE),
+	/* GPP_C10 - GPIO */
+	PAD_NC(GPP_C10, NONE),
+	/* GPP_C11 - GPIO */
+	PAD_NC(GPP_C11, NONE),
+	/* GPP_C12 - GPIO */
+	PAD_NC(GPP_C12, NONE),
+	/* GPP_C13 - GPIO */
+	PAD_NC(GPP_C13, NONE),
+	/* GPP_C14 - GPIO */
+	PAD_NC(GPP_C14, NONE),
+	/* GPP_C15 - GPIO */
+	PAD_NC(GPP_C15, NONE),
+	/* GPP_C16 - GPIO */
+	PAD_NC(GPP_C16, NONE),
+	/* GPP_C17 - GPIO */
+	PAD_NC(GPP_C17, NONE),
+	/* GPP_C18 - GPIO */
+	PAD_NC(GPP_C18, NONE),
+	/* GPP_C19 - GPIO */
+	PAD_NC(GPP_C19, NONE),
+	/* GPP_C20 - GPIO */
+	PAD_NC(GPP_C20, NONE),
+	/* GPP_C21 - GPIO */
+	PAD_NC(GPP_C21, NONE),
+	/* GPP_C22 - GPIO */
+	PAD_NC(GPP_C22, NONE),
+	/* GPP_C23 - GPIO */
+	PAD_NC(GPP_C23, NONE),
+
+	/* ------- GPIO Group GPP_F ------- */
+
+	/* GPP_F0 - GPIO */
+	PAD_NC(GPP_F0, NONE),
+	/* GPP_F1 - GPIO */
+	PAD_NC(GPP_F1, NONE),
+	/* GPP_F2 - GPIO */
+	PAD_NC(GPP_F2, NONE),
+	/* GPP_F3 - GPIO */
+	PAD_NC(GPP_F3, NONE),
+	/* GPP_F4 - GPIO# */
+	PAD_NC(GPP_F4, NONE),
+	/* GPP_F5 - GPIO */
+	PAD_NC(GPP_F5, NONE),
+	/* GPP_F6 - GPIO */
+	PAD_NC(GPP_F6, NONE),
+	/* GPP_F7 - GPIO */
+	PAD_NC(GPP_F7, NONE),
+	/* GPP_F8 - GPIO */
+	PAD_NC(GPP_F8, NONE),
+	/* GPP_F9 - GPIO */
+	PAD_NC(GPP_F9, NONE),
+	/* GPP_F10 - GPIO */
+	PAD_NC(GPP_F10, NONE),
+	/* GPP_F11 - GPIO */
+	PAD_NC(GPP_F11, NONE),
+	/* GPP_F12 - GPIO */
+	PAD_NC(GPP_F12, NONE),
+	/* GPP_F13 - GPIO */
+	PAD_NC(GPP_F13, NONE),
+	/* GPP_F14 - GPIO */
+	PAD_NC(GPP_F14, NONE),
+	/* GPP_F15 - GPIO */
+	PAD_NC(GPP_F15, NONE),
```

Comment:
Isn't it routed to M.2 slot?

Reply:
Seems like some pin to turn the LTE module on/off. Haven't found any information how this on/off feature works. Until I figure out how it is supposed to work (e.g. in ACPI) I leave it as NC to not interfere with LTE module operation.

Reply:
> Seems like some pin to turn the LTE module on/off.

Exactly. It is a feature we can integrate in Dasharo setup, we already have similar ones for other platforms.

Was it also a planned decision? Why not at the very least leave a comment that it looks like a LTE switch and can be investigated later?

> Until I figure out how it is supposed to work (e.g. in ACPI) I leave it as NC to not interfere with LTE module operation.

And how did you plan to keep track of it?

Reply:
LTE switch was never seen on nay other board by me. It might not be even needed at all.

> And how did you plan to keep track of it?

No plans. The LTE module will work without this GPIO. I don't have time to pursue all nice-to-haves with little merit.

Reply:
Added a comment with TODO